### PR TITLE
Extract excludes into a file, fix the link checker by adding http://site.icu-project.org/.

### PR DESCRIPTION
**Pull request by dblock**
Signed-off-by: dblock <dblock@dblock.org>

### Description

- Fixes the link checker failure we're seeing on main by excluding http://site.icu-project.org/ that is currently MIA.
- Extract `--exclude` config into a file.
- Remove now working or obsolete links from the excludes.
 
### Check List
- [x] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by dblock** [comment]
start gradle check

**Review by VachaShah — APPROVED** [review]
LGTM. Thank you @dblock!

File: .lycheeexclude-mail [hunk 1]
```diff
@@ -0,0 +1,4 @@
+"**/*.html"
```
Comment (VachaShah) [inline .lycheeexclude-mail:1]: `--exclude-mail` is just a flag to exclude checking mailing addresses. The `"**/*.html" "**/*.md" "**/*.txt" "**/*.json"` are the files that need to be checked.

**Comment by nknize** [comment]
Not sure what lychee is but is this addressing the ongoing linkchecker failure?

**Comment by dblock** [comment]
> Not sure what lychee is but is this addressing the ongoing linkchecker failure?

It's the link checker. One of the websites, http://site.icu-project.org/, has gone MIA, so it started failing. This addresses that + makes it easier to address it in the future by extracting the configuration into config files.

**Comment by dblock** [comment]
start gradle check

File: .lycheeexclude-mail (hunk 1 above)
Comment (dblock) [reply .lycheeexclude-mail:1]: 🤦 my bad

**Comment by dblock** [comment]
start gradle check

**Comment by dblock** [comment]
Can I please have another CR @VachaShah after fixing my epic fail of `--exclude-mail`:)

**Review by abbashus — APPROVED** [review]

**Review by saratvemulapalli — APPROVED** [review]

**Pull request merged**